### PR TITLE
feat(81): add signUp verification with code, Android

### PR DESCRIPTION
**Pull request by 0to1to0to1**

### CODE REVIEW

**Review by dartwlad — APPROVED** [review]

File: packages/mobile/src/components/FlowerForms/FlowerFormDelete.tsx [hunk 1]
```diff
@@ -9,21 +9,22 @@ import {
...
 }
 
 export const FlowerFormDelete: FC<ShelfFormConfigProps> = ({
-  shelf = {},
+  flowerId,
+  shelfId,
   closeFunc,
 }) => {
-  const { id, shelfId } = shelf;
+  // const { id, shelfId } = shelf;
```
Comment (oleksiydubovykTR) [inline packages/mobile/src/components/FlowerForms/FlowerFormDelete.tsx:27]: So remove it maybe?

File: packages/mobile/src/pages/ScreensAuthorization/SignIn/SignIn.tsx [hunk 2]
```diff
@@ -26,6 +26,7 @@ import { NotifyMessage } from './../../../components/NotifyMessage';
 
 const signInBg = require('./../../../assets/img/bg-signIn.png');
 
+// TODO: Should remove memory lick
```
Comment (oleksiydubovykTR) [inline packages/mobile/src/pages/ScreensAuthorization/SignIn/SignIn.tsx:29]: That's a nasty spelling, man :)

File: packages/mobile/src/pages/ScreensAuthorization/SignIn/SignIn.tsx [hunk 3]
```diff
@@ -148,8 +150,8 @@ class SignInComponent extends React.Component<AuthorizationProps, SignInState> {
 }
 
 const mapStateToProps = ({ auth }: InterfaceStore) => {
-  const { error } = auth;
-  return { error };
+  const { error, mobileToken } = auth;
+  return { error, mobileToken };
```
Comment (oleksiydubovykTR) [inline packages/mobile/src/pages/ScreensAuthorization/SignIn/SignIn.tsx:154]: you could
return { ...auth }

File: packages/mobile/src/pages/ScreensAuthorization/SignUp/SignUp.tsx [hunk 4]
```diff
@@ -277,8 +273,8 @@ export class SignUpComponent extends React.Component<
 }
 
 const mapStateToProps = ({ auth }: InterfaceStore) => {
-  const { error, info, isEmailVerification } = auth;
-  return { error, info, isEmailVerification };
+  const { error, info, isEmailVerification, mobileToken } = auth;
+  return { error, info, isEmailVerification, mobileToken };
```
Comment (oleksiydubovykTR) [inline packages/mobile/src/pages/ScreensAuthorization/SignUp/SignUp.tsx:277]: you could
return { ...auth }

File: packages/mobile/src/pages/ScreensAuthorization/VerificationEmailWithCode/VerificationEmailWithCode.tsx [hunk 5]
```diff
@@ -11,59 +11,52 @@ import {
...
-  const [code, setCode] = useState('');
-  const [validationStatus, setValidationStatus] = useState({
-    status: null,
-    isShowMessage: false,
-  });
+  const dispatch = useDispatch();
+  const [codeLetterFirst, setCodeLetterFirst] = useState('');
+  const [codeLetterSecond, setCodeLetterSecond] = useState('');
+  const [codeLetterThird, setCodeLetterThird] = useState('');
+  const [codeLetterFourth, setCodeLetterFourth] = useState('');
```
Comment (oleksiydubovykTR) [inline packages/mobile/src/pages/ScreensAuthorization/VerificationEmailWithCode/VerificationEmailWithCode.tsx:24]: That's an interesting use of useStates :)

File: packages/mobile/src/pages/UserScreens/Shelves/Shelves.tsx [hunk 6]
```diff
@@ -17,18 +18,25 @@ import { ShelfFormConfig } from '../../../components/ShelfForms';
...
+  if (!shelves || !shelves.length) {
     return (
-      <TouchableOpacity
-        onPress={() => navigation.navigate(SCREEN_USER_SHELF, { ...shelf })}
-        style={styles.shelves__item}
-        key={shelf.id}>
-        <Shelf {...shelf} />
-      </TouchableOpacity>
+      <View>
+        <Text>Sheves are empty</Text>
```
Comment (oleksiydubovykTR) [inline packages/mobile/src/pages/UserScreens/Shelves/Shelves.tsx:24]: typo

**Review by oleksiydubovykTR — APPROVED** [review]

**Review by dubbha — APPROVED** [review]

**Pull request merged**